src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1603,6 +1618,9 @@ public void marshal(Object source, HierarchicalStreamWriter w, MarshallingContex
                 if (e.timings != null) {
                     writeChild(w, context, "timings", e.timings, Map.class);
                 }
+                if (e.internalCalls != null) {
+                    writeChild(w, context, "internalCalls", e.internalCalls, Set.class);
```

Comment:
Potentially clever solution to avoid use of synchronization: de/serialize to/from a trivial Set implementation, but use a thread-safe representation (such as ConcurrentHashMap-backed impls) in-memory.

Reply:
Again, probably overkill for an infrequently-modified collection.